### PR TITLE
Add pnpm-sync-dependencies-meta-injected to improve DX

### PR DESCRIPTION
**Pull request by ynotdraw**
## 🚀 Description

This PR follows @vitch 's `headless-form` PR to make local development a lot better.  Instead of copying over what's there, I'm instead using @NullVoxPopuli 's [package](https://github.com/NullVoxPopuli/pnpm-sync-dependencies-meta-injected) that does this for us.

With these changes, instead of doing `pnpm` linking or running a script for local development, run `pnpm start` from the root of the monorepo. When making a change, the packages will rebuild and your change will be displayed.

- https://github.com/CrowdStrike/ember-headless-form/pull/140

---

## 🔬 How to Test

- Pull this branch down
- Run `pnpm i`
- Run `pnpm start`
- Verify the docs and test apps start
- Make a change under `ember-toucan-core` and `ember-toucan-form`
- Verify the change shows up in the docs/test app

---

## 📸 Images/Videos of Functionality

N/A - developer improvement only

### CODE REVIEW

File: local-setup.bash
Comment (nicolechung) [inline local-setup.bash:1]: Nice to see this deleted!

**Review by simonihmig — APPROVED** [review]
Nice!

**Comment by nicolechung** [comment]
I did
```
pnpm i
pnpm start
```

And got this:

Screenshot 2023-06-28 at 9 24 10 AM

1. Is an initial build still required?
2. If so, should we add that instruction to the docs?

**Comment by nicolechung** [comment]
> I did
> 
> ```
> pnpm i
> pnpm start
> ```
> 
> And got this:
> 
> Screenshot 2023-06-28 at 9 24 10 AM
> 1. Is an initial build still required?
> 2. If so, should we add that instruction to the docs?

Update: after running `pnpm start` the 2nd time it worked. I do think there should be a note in the docs about this though because I was running from a totally clean install (no `dist` folder)

**Pull request merged**